### ISSUE TITLE
Crop activity styling

### ISSUE THREAD

**Issue by alex-shpak** [jdamcd/android-crop#119]
Hi!
Is there way to customize cropping activity?
e.g. Color of "actionBar" and color of border of selector?

**Comment by urizev** [comment]
I agree with you. I would like to change the color of the bar and move it to the bottom.

**Comment by jdamcd** [comment]
The colour of the area selector is customisable. Check out the theme in the example project.

**Comment by sridharananthakrishnan** [comment]
Yes, Alexander ... Its possible . .. You just use android support library
in your project . Anyways you have opensource library of SoundCloud/crop
library ...u just go through it ... U can use it on ur own taste ...if u
need any help ...ask me ..I'll help u

**Comment by nfegard** [comment]
I would love for someone to direct me to the area where the styling options are listed within the library. I tampered with the options provided in the example but I still feel like the cropping activity could use a color change to fit my app better.

**Comment by alex-shpak** [comment]
Ok. I found style for highlight, thanks.
It should be enough for the beginning.

@jdamcd 
What do you think about using Theme and attribute references to customize.
I can take a look into it.

**Comment by alex-shpak** [comment]
@nfegard 
Looks like the only option for customization now is adding
`<item name="cropImageStyle">@style/Widget.CropImageView</item>`
to your app theme, where `Widget.CropImageView` is
```xml
<style name="Widget.CropImageView" parent="android:Widget">
        <item name="showThirds">true</item>
        <item name="showCircle">false</item>
        <item name="showHandles">always</item>
        <item name="highlightColor">@color/highlight</item>
</style>
```

Look [here](https://github.com/jdamcd/android-crop/blob/master/example/src/main/res/values/theme.xml) for example
And [here](https://github.com/jdamcd/android-crop/blob/master/lib/src/main/res/values/attrs.xml) for available values for attributes.

**Comment by nfegard** [comment]
I have found these so far:
<item name="showThirds">true</item>
<item name="showHandles">always</item>
<item name="highlightColor">custom_color</item>
<item name="Crop.ActionButtonText.Cancel">custom_cancel_button_style</item>
<item name="Crop.ActionButtonText.Done">custom_done_button_style</item>

**Comment by nfegard** [comment]
Is there one to change the background???

**Comment by nfegard** [comment]
![image](https://cloud.githubusercontent.com/assets/8585983/7770152/f850f31a-003f-11e5-95ad-73a753260497.png)

**Comment by nfegard** [comment]
This is the style file and there is no background option. poop.

![image](https://cloud.githubusercontent.com/assets/8585983/7770175/4b7a6210-0040-11e5-99af-44a8f85f5503.png)

**Comment by nfegard** [comment]
So here is an idea, since we know the names of the drawables used to define the look of the background and the buttons according to this stack overflow answer:

As long as the resources have the same filename, the ones defined in the app will override the ones from the lib.

**Comment by alex-shpak** [comment]
Yes, you can define resource (drawable or color) with the same name in your app and it should work.

**Comment by jdamcd** [comment]
Yep. All the resources can be overridden in your own project. However, the resource IDs might change without notice in future releases.

@nfegard I think the background you're looking for is `@color/crop__button_bar`?

I'll have a think about how to support this better beyond just adding a bunch of custom attributes. `Toolbar` & `AppCompat` themes is one option, but I'm reluctant to add that dependency.

**Comment by alex-shpak** [comment]
I think adding `AppCompat` as dependency is not good idea. Maybe add style reference attributes, like
`?attr:cropButtonStyleOk`, `?attr:cropButtonStyleCancel` where user can define fully new style, or use button style from his app theme, or extend from library default style.

**Comment by PriyankPainter** [comment]
Q. When i get image from Camera and Gallery alternatively then it works perfectly but when i want to replace image coming from gallary to new gallary image then it not replace my image?

it displays crop but not replace image in ImageView, help as soon as possible.
